mypy/semanal.py

```diff
@@ -3024,12 +3039,9 @@ def infer_if_condition_value(expr: Node, pyversion: Tuple[int, int], platform: s
         elif name == 'PY3':
             result = ALWAYS_TRUE if pyversion[0] == 3 else ALWAYS_FALSE
         elif name == 'MYPY' or name == 'TYPE_CHECKING':
-            result = ALWAYS_TRUE
+            result = MYPY_TRUE
```

Comment:
Well, we don't really care about other type checkers in this codebase, do we? :-)